### PR TITLE
Firefly-1571: support image sub-highlighting

### PR DESCRIPTION
**Pull request by robyww**
#### Review
@loitly can you check the `TableUtil.js` changes?

#### Testing
- https://firefly-1571-image-sub-highlighting.irsakudev.ipac.caltech.edu/applications/spherex
- you should see the image sub-highlight like the table does

### CODE REVIEW

File: src/firefly/js/visualize/iv/ImageViewerDecorate.jsx [hunk 1]
```diff
@@ -282,18 +282,19 @@ const ImageViewerDecorate= memo((props) => {
...
         const manyPlots= hasManyPlots(pv,visRoot);
+        const active= isActivePlotView(visRoot,pv.plotId);
         return {
             width: !manyPlots  ? 1 : 'calc(100% - 4px)',
             bottom: 0,
             top: 0,
             overflow: 'hidden',
             position: 'absolute',
-            borderStyle: manyPlots ? 'solid' : undefined,
+            borderStyle: !manyPlots ? undefined : (pv.subHighlight && !active) ? 'dashed' : 'solid' ,
```
Comment (jaladh-singhal) [inline src/firefly/js/visualize/iv/ImageViewerDecorate.jsx:292]: I like the dashed style

File: src/firefly/js/visualize/saga/ActiveRowToImageWatcher.js [hunk 2]
```diff
@@ -20,53 +22,63 @@ import CoordinateSys from '../CoordSys';
...
 
 
+
 function willFitOnScreenAtCurrentZoom(pv) {
     const {w,h}=computeBoundingBoxInDeviceCoordsForPlot(primePlot(pv)) ?? {};
     return (pv.viewDim.width+3>=w && pv.viewDim.height+3>=h);
 }
 
+const getTableActions= () =>  [TABLE_LOADED, TABLE_SELECT, TABLE_HIGHLIGHT, TABLE_UPDATE, TBL_RESULTS_ACTIVE, TABLE_REMOVE];
+const getImageActions= () =>  [ImagePlotCntlr.CHANGE_ACTIVE_PLOT_VIEW, ImagePlotCntlr.ANY_REPLOT];
```
Comment (jaladh-singhal) [inline src/firefly/js/visualize/saga/ActiveRowToImageWatcher.js:32]: not sure why these are defined as functions and not just variables?

File: src/firefly/js/tables/TableUtil.js [hunk 3]
```diff
@@ -999,6 +999,42 @@ export function setHlRowByRowIdx(nreq, tableModel) {
...
+
+    const relatedColsHL = relatedCols
+        .split(',')
+        .map((cname) => getCellValue(tableModel, highlightedRow, cname?.trim()))
+        .join('|');
+
+    const cVal = relatedCols
+        .split(',')
+        .map((cname) => getCellValue(tableModel, rowIdx, cname?.trim()))
+        .join('|');
```
Comment (jaladh-singhal) [inline src/firefly/js/tables/TableUtil.js:1026]: this chained string manipulation can be defined as an inline function with row as a param, and then can be reused in both cases.

**Review by jaladh-singhal — APPROVED** [review]
Works as expected! 

One minor thing: the sub-highlights on image (dashed border) doesn't appear until Image is rendered, but the highlight (solid border) does. This becomes noticeable when you quickly change the row highlight to the bottom of table, and there's lag of 10sec from "loading image" state to rendered state.

File: src/firefly/js/visualize/saga/ActiveRowToImageWatcher.js (hunk 2 above)
Comment (robyww) [reply src/firefly/js/visualize/saga/ActiveRowToImageWatcher.js:32]: because of cross import issues

**Review by loitly — APPROVED** [review]
Looks good.  However, I noticed it crashes easily with 2GB deployment.
I did a LVF search, then switch to image grid mode, showing 8 images.  I did another search, again switch to grid mode.  When I switch back to the previous results tab, the POD crashes.  I was able to duplicate it again doing the exact same thing.
Maybe we should do a load test before releasing it to the public.  It's okay if it's slow, but it should not crash.

**Pull request merged**